### PR TITLE
Changes made to plot behaviour of loss of the mode lwhen no of epoch …

### PR DESCRIPTION
**Pull request by Bekal-Ramnath-Rao**
…is changed

### CODE REVIEW

**Review by shaolong5 — APPROVED** [review]

File: Project.toml [hunk 1]
```diff
@@ -5,11 +5,13 @@ version = "1.0.0-DEV"
 
 [deps]
 DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
+GLMakie = "e9467ef8-e4e7-5192-8a1a-b1aee30e663a"
```
Comment (behinger) [inline Project.toml:8]: the plotting library GLMakie should not be a dependency of thepackage. You could use CairoMakie in the docs - or have your own enviromnemt somewhere to add plotting libraries

File: Project.toml [hunk 2]
```diff
@@ -5,11 +5,13 @@ version = "1.0.0-DEV"
...
 Lux = "b2108857-7c20-44ae-9111-449ecde12c47"
 LuxCUDA = "d0bbae9a-e099-4d5b-a835-1c6931763bda"
 Optimisers = "3bd65402-5787-11e9-1adc-39752487f4e2"
 ProgressMeter = "92933f4c-e287-5a05-a399-4b506db050ca"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Revise = "295af30f-e4ad-537b-8983-00126c2a3abe"
```
Comment (behinger) [inline Project.toml:14]: the same for revise - it typically is even in the global environment

File: docs/literate/main.jl [hunk 3]
```diff
@@ -6,7 +6,9 @@ using CairoMakie
 using Statistics
 using LuxCUDA
 using Revise
+using testdata
```
Comment (behinger) [inline docs/literate/main.jl:9]: there is no package `testdata` that I'm aware of - are you sure this is what you need?

File: docs/literate/main.jl [hunk 4]
```diff
@@ -6,7 +6,9 @@ using CairoMakie
...
 includet("../testdata.jl")
+# Create an object of the MersenneTwister, use this object to create random number between 0 and 1
```
Comment (behinger) [inline docs/literate/main.jl:11]: to clarify: MersenneTwister is an algorithm to generate random numbers, but the actual random number function to create numbers between 0 and 1 is simply `rand`

File: src/traintest.jl [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
 function train(dre::DRE, eeg_in, eeg_out, ps, st; n_epochs=1, lr=0.01, batch_size=32, show_progress=true)
-    @info "Input data is a $(typeof(eeg_in)) and $(typeof(eeg_out)) with size $(size(eeg_in)),$(size(eeg_out))"
 
+    @info "Input data is a $(typeof(eeg_in)) and $(typeof(eeg_out)) with size $(size(eeg_in)),$(size(eeg_out))"
     # conv layer expects input of shape (time, channel, epoch)
     opt_state = create_optimiser(ps, lr)
     p = Progress(n_epochs)
+    
+    loss_epoch_array = Float64[]
+    loss_epoch_array = Array{Float64}(undef,n_epochs)
```
Comment (behinger) [inline src/traintest.jl:10]: you do not need the first line here, as it is immediately overwritten by the second one

**Pull request merged**